usaspending_api/financial_activities/models.py

```diff
@@ -56,3 +56,165 @@ class FinancialAccountsByProgramActivityObjectClass(DataSourceTrackedModel):
+    gross_outlays_delivered_orders_paid_total_cpe = models.DecimalField(max_digits=21, decimal_places=2)
+    gross_outlay_amount_by_program_object_class_fyb = models.DecimalField(max_digits=21, decimal_places=2)
+    gross_outlay_amount_by_program_object_class_cpe = models.DecimalField(max_digits=21, decimal_places=2)
+    obligations_incurred_by_program_object_class_cpe = models.DecimalField(max_digits=21, decimal_places=2)
+    ussgl487100_down_adj_pri_unpaid_undel_orders_oblig_recov_cpe = models.DecimalField(max_digits=21, decimal_places=2)
+    ussgl497100_down_adj_pri_unpaid_deliv_orders_oblig_recov_cpe = models.DecimalField(max_digits=21, decimal_places=2)
+    ussgl487200_down_adj_pri_ppaid_undel_orders_oblig_refund_cpe = models.DecimalField(max_digits=21, decimal_places=2)
+    ussgl497200_down_adj_pri_paid_deliv_orders_oblig_refund_cpe = models.DecimalField(max_digits=21, decimal_places=2)
+    deobligations_recoveries_refund_pri_program_object_class_cpe = models.DecimalField(max_digits=21, decimal_places=2)
+    create_date = models.DateTimeField(auto_now_add=True, blank=True, null=True)
+    update_date = models.DateTimeField(auto_now=True, null=True)
+
+    class Meta:
+        managed = True
+        db_table = 'tas_program_activity_object_class_quarterly'
+        unique_together = (
+            'treasury_account', 'program_activity', 'object_class', 'submission')
+
+    @classmethod
+    def insert_quarterly_numbers(cls, submission_id=None):
+        """
+        Bulk insert quarterly finanical numbers by tas, program activity, and
+        object class.
+        """
+        field_list = [field.column for field in TasProgramActivityObjectClassQuarterly._meta.get_fields()]
+
+        # delete existing quarterly data
+        if submission_id is None:
+            TasProgramActivityObjectClassQuarterly.objects.all().delete()
+        else:
+            TasProgramActivityObjectClassQuarterly.objects.filter(
+                submission_id=submission_id).delete()
+
+        # retrieve RawQuerySet of quarterly breakouts
+        qtr_records = FinancialAccountsByProgramActivityObjectClass.get_quarterly_numbers(
+            submission_id)
+        qtr_list = []
+
+        # for each record in the RawQuerySet, create a corresponding
+        # TasProgramActivityObjectClassQuarterly object and save it to a list
+        # for subsequent bulk insert
+        # TODO: maybe we don't want this entire list in memory?
+        for rec in qtr_records:
+
+            # remove any fields in the qtr_record RawQuerySet object so that we
+            # can create its TasProgramActivityObjectClassQuarterly counterpart
+            # more easily. it's a bit hacky, but avoids the need to explicitly
+            # set dozens of attributes when creating TasProgramActivityObjectClassQuarterly
+            rec_dict = rec.__dict__
+            for key in list(rec_dict.keys()):
```

Comment:
I think you could boild these three lines down to 

`rec_dict = {key: rec_dict[key] for key in rec_dict if key in field_list}`

if you love comprehensions as gratuitously as I do.